metadata-ingestion/tests/integration/azure_ad/test_azure_ad.py

```diff
@@ -107,6 +107,61 @@ def test_azure_ad_source_default_configs(pytestconfig, tmp_path):
     )
 
 
+@freeze_time(FROZEN_TIME)
+def test_azure_ad_source_empty_group_membership(pytestconfig, tmp_path):
+    test_resources_dir: pathlib.Path = (
+        pytestconfig.rootpath / "tests/integration/azure_ad"
+    )
+
+    with patch(
+        "datahub.ingestion.source.identity.azure_ad.AzureADSource.get_token"
+    ) as mock_token, patch(
+        "datahub.ingestion.source.identity.azure_ad.AzureADSource._get_azure_ad_users"
+    ) as mock_users, patch(
+        "datahub.ingestion.source.identity.azure_ad.AzureADSource._get_azure_ad_groups"
+    ) as mock_groups, patch(
+        "datahub.ingestion.source.identity.azure_ad.AzureADSource._get_azure_ad_group_members"
+    ) as mock_group_users:
+        mocked_functions(
+            test_resources_dir, mock_token, mock_users, mock_groups, mock_group_users
+        )
+        # Run an azure usage ingestion run.
+        pipeline = Pipeline.create(
+            {
+                "run_id": "test-azure-ad",
+                "source": {
+                    "type": "azure-ad",
+                    "config": {
+                        "client_id": "00000000-0000-0000-0000-000000000002",
+                        "tenant_id": "00000000-0000-0000-0000-000000000002",
+                        "client_secret": "client_secret",
+                        "redirect": "https://login.microsoftonline.com/common/oauth2/nativeclient",
+                        "authority": "https://login.microsoftonline.com/00000000-0000-0000-0000-000000000000",
+                        "token_url": "https://login.microsoftonline.com/00000000-0000-0000-0000-000000000000/oauth2/token",
+                        "graph_url": "https://graph.microsoft.com/v1.0",
+                        "ingest_group_membership": True,
+                        "ingest_groups": True,
+                        "ingest_users": True,
+                    },
+                },
```

Comment:
if possible, could we use a pytest parameterized test

Reply:
wdym?

Reply:
or do you have an example i can look at?